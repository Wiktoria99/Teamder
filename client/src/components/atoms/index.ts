export * from './BackButton';
export * from './CustomTextField';
export * from './CustomButton';
export * from './SideBarButton';
