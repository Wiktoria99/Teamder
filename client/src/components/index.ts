export * from "./atoms";
export * from "./atoms";
export * from "./molecules";
export * from "./organisms";
