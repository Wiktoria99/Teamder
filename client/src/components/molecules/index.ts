export * from './RegisterSteps';
export * from './TeamItem';
export * from './Suggestions';
export * from './Sidebar';
export * from './TeamList';
export * from './SuggestionItem';
export * from './SuggestionList';
export * from './NotificationItem';
export * from './NotificationList';
