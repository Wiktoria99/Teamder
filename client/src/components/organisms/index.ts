export * from './Layout';
export * from './FormWrapper';
export * from './MainWrapper';
export * from './BarWrapper';
