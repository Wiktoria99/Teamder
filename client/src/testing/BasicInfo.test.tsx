import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { BasicInfo, CustomTextField, CustomButton } from '@/components';
import { cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(cleanup);

const request = {
  user_name: 'test',
  password: 'test',
  password2: 'test',
  name: 'test',
  surname: 'test',
  email: 'test',
  location: 'test',
  age: 2,
  photo_src: 'test',
  bio: 'test',
  social_media_URL1: 'test',
  social_media_URL2: 'test',
  social_media_URL3: 'test',
  list_of_interests: ['test'],
};

it('BasicInfo renders without crashing', () => {
  shallow(<Router><BasicInfo registerInfo={request} setRegisterInfo='' updateProgressState=''/></Router>);
});

it('BasicInfo matches snapshot', () => {
  const tree = shallow(<Router><BasicInfo registerInfo={request} setRegisterInfo='' updateProgressState=''/></Router>);
  expect(toJson(tree)).toMatchSnapshot();
});

it("find button and textfields", () => {
  const wrapper = mount(<Router><BasicInfo registerInfo={request} setRegisterInfo='' updateProgressState=''/></Router>);
  expect(wrapper.find(CustomTextField).length).toBe(8);
  expect(wrapper.contains(CustomButton)).toBe(true);
});