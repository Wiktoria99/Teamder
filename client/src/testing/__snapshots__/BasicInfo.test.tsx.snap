// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInfo matches snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <BasicInfo
    registerInfo={
      Object {
        "age": 2,
        "bio": "test",
        "email": "test",
        "list_of_interests": Array [
          "test",
        ],
        "location": "test",
        "name": "test",
        "password": "test",
        "password2": "test",
        "photo_src": "test",
        "social_media_URL1": "test",
        "social_media_URL2": "test",
        "social_media_URL3": "test",
        "surname": "test",
        "user_name": "test",
      }
    }
    setRegisterInfo=""
    updateProgressState=""
  />
</Router>
`;
